src/Cli/dotnet/Commands/Run/RunCommand.cs

```diff
@@ -239,13 +234,9 @@ internal bool TryGetLaunchProfileSettingsIfNeeded(out ProjectLaunchSettingsModel
 
         return true;
 
-        static string? TryFindLaunchSettings(string projectOrEntryPointFilePath)
+        static string? TryFindLaunchSettings(string projectOrEntryPointFilePath, string? launchProfile)
         {
-            var buildPathContainer = File.Exists(projectOrEntryPointFilePath) ? Path.GetDirectoryName(projectOrEntryPointFilePath) : projectOrEntryPointFilePath;
-            if (buildPathContainer is null)
-            {
-                return null;
-            }
+            var buildPathContainer = File.Exists(projectOrEntryPointFilePath) ? Path.GetDirectoryName(projectOrEntryPointFilePath)! : projectOrEntryPointFilePath;
```

Comment:
I am struggling a bit to follow this File.Exists check. I realize it is pre-existing code. But what is this doing? Is this really saying: if this is a file, then use its containing directory, otherwise assume it's a directory and use the path as-is? Is that something we should verify (e.g. `buildPathContainer` should always be a directory on disk)?